cmd/internal/compress/compress.go

```diff
@@ -0,0 +1,93 @@
+package compress
+
+import (
+	"fmt"
+	"path/filepath"
+
+	"github.com/metal-stack/backup-restore-sidecar/cmd/internal/constants"
+	"github.com/mholt/archiver/v3"
+)
+
+type (
+	// Compressor is responsible to compress and decompress backups
+	Compressor interface {
+		// Compress the given backupFile and returns the full filename with the extension
+		Compress(backupFilePath string) (string, error)
+		// Decompress the given backupFile
+		Decompress(backupFilePath string) error
+	}
```

Comment:
I find the interface approach confusing because there is only one single struct that implements the interface and the single struct supports all the methods.

Shouldn't there be TarCompressor, a TarGZCompressor, ... structs that all implement the Compressor interface? If not, I think we can just delete the interface.